pkg/ibmcsidriver/server.go

```diff
@@ -158,3 +160,42 @@ func logGRPC(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, h
 	}
 	return resp, err
 }
+
+func removeCSISocket(endPoint string) {
+	// Reference: https://github.com/kubernetes-csi/node-driver-registrar/blob/master/cmd/csi-node-driver-registrar/node_register.go#L168
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, syscall.SIGTERM)
+	<-sigc
+	err := os.Remove(endPoint)
+	if err != nil && !os.IsNotExist(err) {
+		glog.Errorf("failed to remove socket: %s with error: %+v", endPoint, err)
+	}
+	/*
+		This is a temporary code to cleanup csi-socket created under csi-plugins directory.
+		This code must be removed once current supported versions are deprecated and
+		new major release is done.
+	*/
+	csiPluginDataPath := "/kubelet-data-dir/csi-plugins/vpc.block.csi.ibm.io/"
+	csiPluginLibPath := "/kubelet-lib-dir/csi-plugins/vpc.block.csi.ibm.io/"
+	directoryDelete(csiPluginDataPath)
+	directoryDelete(csiPluginLibPath)
+	os.Exit(0)
+
+}
+
+func directoryDelete(csiPluginSocketPath string) {
+	if directoryExists(csiPluginSocketPath) {
+		err := os.RemoveAll(csiPluginSocketPath)
```

Comment:
no need to verify directoryExists or not as its good to have and we will remove code later

Reply:
@ambiknai `directoryExists` is check is not needed as @arahamad mentioned because RemoveAll does not return any error if Directory does not exist.

`RemoveAll removes path and any children it contains. It removes everything it can but returns the first error it encounters. If the path does not exist, RemoveAll returns nil (no error). If there is an error, it will be of type *PathError.`

Reply:
Yes Sushma. This is taken care